### DISCUSSION TITLE
自定义产出目录&输出配置

### DISCUSSION THREAD

**Discussion by leaf-potato** [xmake-io/xmake#3389] [Q&A]
请教下，想把binary的target放到bin目录，library（static shared）放到lib目录怎么配置呢
```
.
├── bin
├── conf
└── lib
```
以及想把配置文件也输出到output目录里，是使用哪个标签呢

**Comment by star-hengxing** [comment]
试一下`xmake install -o xxx`，会打包好整个库到 xxx 目录
如果不满足需求可以自定义 on_install

**Reply by leaf-potato** [reply]
这个是打包操作耶，现在想在`xmake build`时候输出到不同目录

**Comment by leaf-potato** [comment]
xmake编译后将产出放到`${outputdir}/${platform}/${architecture}/${mode}`目录下， 想根据不同产出输出到不同目录下，类似cmake中这种标签：

``` cmake
# 设置lib的输出路径
set(LIBRARY_OUTPUT_PATH ${PROJECT_SOURCE_DIR}/lib)
```

**Reply by star-hengxing** [reply]
https://xmake.io/#/zh-cn/manual/project_target?id=targetset_targetdir

**Reply by leaf-potato** [reply]
有全局标签可以针对不同 target kind设置的吗

**Reply by star-hengxing** [reply]
可以写一个 rule 里根据 kind 设置 targetdir，然后给每个 target 加上这个 rule

**Reply by leaf-potato** [reply]
好的 感谢~ 但感觉这个功能比较常用 能否考虑内置支持呢

**Reply by leaf-potato** [reply]
想问下 如果rule想对所有的target生效（不挨个添加） 该如何操作呢？好像在全局域使用`add_rules`是不行的

**Comment by leaf-potato** [comment]
以及某些配置文件在`xmake build`时想放到`outputdir`目录，这个有标签支持吗？在文档里没找到说明耶

**Reply by star-hengxing** [reply]
https://xmake.io/#/zh-cn/manual/project_target?id=targetafter_build
在 after build 里 `os.cp("xxx",  target:targetdir())`

**Reply by waruqi** [reply]
也可以直接用这个

https://xmake.io/#/zh-cn/manual/project_target?id=targetadd_configfiles
https://xmake.io/#/zh-cn/manual/project_target?id=targetset_configdir

**Reply by leaf-potato** [reply]
收
